### PR TITLE
Add CONTRIBUTING.md and update README.md

### PR DESCRIPTION
**Pull request by youngnick**
Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# Contributing
+
+Thanks for taking the time to join our community and start contributing.
+These guidelines will help you get started with the Contour project.
+Please note that we require [DCO sign off](#dco-sign-off).  
+
+## Building from source
+
+This section describes how to build Contour from source.
```
Comment (jpeach) [inline CONTRIBUTING.md]: s/Contour/ir2proxy/

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+### Prerequisites
+
+1. *Install Go*
+
+    ir2proxy requires [Go 1.13][1] or later.
+    We also assume that you're familiar with Go's [`GOPATH` workspace][3] convention, and have the appropriate environment variables set.
```
Comment (jpeach) [inline CONTRIBUTING.md:16]: You don't need GOPATH because modules, right?

Comment (jpeach) [reply CONTRIBUTING.md:16]: Which means that most of the verbiage below is also unnecessary?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+that they are only submitting work that they have rights to, we are requiring
+everyone to acknowledge this by signing their work.
+
+Any copyright notices in this repository should specify the authors as "The
+project authors".
+
+To sign your work, just add a line like this at the end of your commit message:
+
+```
+Signed-off-by: David Cheney <cheneyd@vmware.com>
```
Comment (jpeach) [inline CONTRIBUTING.md]: s/David Cheney/Nick Young/

:)

**Review by jpeach — APPROVED** [review]
LGTM other than a couple of nits. You could elide all the GOPATH material.

File: CONTRIBUTING.md (hunk 2 above)
Comment (youngnick) [reply CONTRIBUTING.md:16]: Yeah, probably. But I lifted it from Contour's one, which hasn't been updated in ages. I'll remove it.

Comment (youngnick) [reply CONTRIBUTING.md:16]: I changed things around instead.

File: CONTRIBUTING.md (hunk 1 above)
Comment (youngnick) [reply CONTRIBUTING.md]: Thanks, there were a couple of other ones too, fixed now.

**Pull request merged**